### PR TITLE
Support minlength argument to np.bincount

### PR DESCRIPTION
**Pull request by AndrewEckart**
Support the minlength argument in np.bincount, update the docs, and add tests.

Apologies for the duplicate PR (#5759), I messed up my version control a little. No rush, I understand the reviewers are quite busy with the RC.

### CODE REVIEW

**Comment by esc** [comment]
@AndrewEckart no problem at all, thanks for cleaning things up and for being mindful! We appreciate it.

File: numba/np/arraymath.py [hunk 1]
```diff
@@ -3308,25 +3311,28 @@ def count_item(out, idx, val, weights):
...
         def count_item(out, idx, val, weights):
             out[val] += 1
 
-    def bincount_impl(a, weights=None):
-        validate_inputs(a, weights)
-        n = len(a)
+    def bincount_impl(a, weights=None, minlength=0):
+        validate_inputs(a, weights, minlength)
+        if minlength < 0:
+            raise ValueError("bincount(): minlength must be non-negative!")
```
Comment (stuartarchibald) [inline numba/np/arraymath.py:3324]:
```suggestion
            raise ValueError("'minlength' must not be negative")
```

File: numba/tests/test_np_functions.py [hunk 2]
```diff
@@ -749,6 +753,37 @@ def test_bincount2_exceptions(self):
...
+        with self.assertRaises(ValueError) as raises:
+            cfunc([2, -1], [0, 0])
+        self.assertIn("first argument must be non-negative",
+                      str(raises.exception))
+
+        # Negative minlength
+        with self.assertRaises(ValueError) as raises:
+            cfunc([17, 38], None, -1)
+        self.assertIn("minlength must be non-negative",
+                      str(raises.exception))
```
Comment (stuartarchibald) [inline numba/tests/test_np_functions.py:785]: probably need to patch this up WRT the changed error message?

Comment (AndrewEckart) [reply numba/tests/test_np_functions.py]: That's what I get for making commits directly from GitHub... should be fixed now.

**Comment by esc** [comment]
@lrdegeest thank you for asking about this, it can take some time between a Numba release and that version becoming available via `conda`.

For what it is worth, the Numba team also produces conda packages in the Numba channel on anaconda.org, which you can install with:

```
conda install -c numba numba
```

**Pull request merged**